libraries/DataFlash/LogFile.cpp

```diff
@@ -1547,12 +1547,12 @@ void DataFlash_Class::Log_Write_AttitudeView(AP_AHRS_View &ahrs, const Vector3f
     WriteBlock(&pkt, sizeof(pkt));
 }
 
-void DataFlash_Class::Log_Write_Current_instance(const AP_BattMonitor &battery,
-                                                 const uint64_t time_us,
+void DataFlash_Class::Log_Write_Current_instance(const uint64_t time_us,
                                                  const uint8_t battery_instance,
                                                  const enum LogMessages type,
                                                  const enum LogMessages celltype)
 {
+    AP_BattMonitor &battery = AP::battery();
```

Comment:
why bother storing the singleton?

Reply:
If you expand the lines below we call a bunch of functions on the battery, it's a bit easier to read without having to prefix it everywhere, and also kept the patch smaller. (I didn't see a lot of value with subsituting `battery` with `AP::battery()` over another 9 lines below)